src/schema/v2/artworksForUser/helpers.ts

```diff
@@ -145,3 +145,21 @@ export const getBackfillArtworks = async (
 
   return (itemsBody || []).slice(0, remainingSize)
 }
+
+export const getDislikedArtworkIds = async (
+  context: ResolverContext
+): Promise<string[]> => {
+  const { collectionArtworksLoader } = context
+
+  if (!collectionArtworksLoader) return []
+
+  const { body: dislikedArtworks } = await collectionArtworksLoader(
```

Comment:
There's no pagination here, what if a user has many (more than 1 page) disliked artworks? We wouldn't be properly filtering them out of recs in that case.

In general (not having looked at the gravity-side implementation) I would think this dislike filter might actually be better off in Gravity, as it can be not-comprehensive (re pagination and large result sets ) to get _all_ dislikes to effectively filter in MP (vs. filtering in Grav) - and if we're going to support filtering out dislikes we should support it comprehensively.

Reply:
> There's no pagination here, what if a user has many (more than 1 page) disliked artworks? We wouldn't be properly filtering them out of recs in that case.

Yes, but we probably don't all of them. The idea is that recs are refreshed daily and users won't see disliked artworks in the end. This PR is a solution for a "temporary" problem - hide disliked artworks until next refresh. We made an assumption that users unlikely will not dislike that many artworks within a day (> then how many artworks collectionArtworksLoader returns).

>  I would think this dislike filter might actually be better off in Gravity

Agree with you, that's why I've started from gravity initially. But then I became hesitant because I need to add the change to 3 different endpoints in gravity and this seems like a big change for this feature.

cc @dblandin

Reply:
> Agree with you, that's why I've started from gravity initially. But then I became hesitant because I need to add the change to 3 different endpoints in gravity and this seems like a big change for this feature.

You mean this approach - https://github.com/artsy/gravity/pull/17558 - added to several endpoints? That actually seems fine/correct to me!

Reply:
Yes, but I will need to add similar behaviour to `filter/artworks` endpoint (called from [here](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/artworksForUser/helpers.ts#L127)) and to `set/:id/items` (from [here](https://github.com/artsy/metaphysics/blob/main/src/schema/v2/artworksForUser/helpers.ts#L144)).

Reply:
Gotcha. Yea, I dunno! I think I'm mildly opinionated that solving something more thoroughly in Gravity is the right way to go, if we want this to be a 'real' feature (that is, we generally try to suppress disliked artworks from some general API endpoints that serve artworks from search/recs/etc. to users). This will work in the future more generally and feels like a nice product addition.

Whereas this MP approach to filter a page of the most recent dislikes from the `artworksForUser` field specifically....it feels a touch too ad-hoc if that makes sense.

It's not a big deal either way so I am 👍 with this MP approach, this is not a blocking comment from me, more raising an alternative/discussion.

Reply:
I think I'm convinced @mzikherman, here is the gravity pr: https://github.com/artsy/gravity/pull/17585 

I think I didn't want to do it "properly" because I've already spent too much time on this ticket and subconsciously I wanted to cut corners, but after re-thinking it I've decided to not introduce technical debt.

Reply:
and updated this PR's implementation

Reply:
Yeah, keeping the implementation in Gravity (where we have better ability to satisfy the request) makes sense to me and the implementation you've proposed in https://github.com/artsy/gravity/pull/17585 seems reasonably straightforward to accomplish our aim here. Thanks for the nudge here @mzikherman 👍🏼